### ISSUE TITLE
ID token verifier hardcodes issuer to accounts.google.com, rejecting IAP tokens

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-rust#6704]
Unconfirmed.  This is the result of a search with Claude, may be a false positive.

In `src/auth/src/credentials/idtoken/verifier.rs`, the expected token issuer is hardcoded:

https://github.com/googleapis/google-cloud-rust/blob/f02b42c2ac5720334cd2b5335824bd4166264527/src/auth/src/credentials/idtoken/verifier.rs#L174-L178

```rust
// TODO(#3591): Support TPC/REP that can have different issuers
validation.set_issuer(&["https://accounts.google.com", "accounts.google.com"]);
```

However, `JwkClient` specifically supports ES256 keys by fetching from the Cloud IAP JWKS URL (`https://www.gstatic.com/iap/verify/public_key-jwk`):

https://github.com/googleapis/google-cloud-rust/blob/f02b42c2ac5720334cd2b5335824bd4166264527/src/auth/src/credentials/internal/jwk_client.rs#L88-L90

Google Cloud Identity-Aware Proxy (IAP) tokens use ES256 signatures with issuer `https://cloud.google.com/iap`. Because the verifier only accepts `accounts.google.com`, all authentic IAP tokens fail verification with `InvalidIssuer`.

**Comment by hartshorne** [comment]
Another concrete case: verifying Cloud IAP's `x-goog-iap-jwt-assertion`. Its issuer is `https://cloud.google.com/iap` and its ES256 keys come from `https://www.gstatic.com/iap/verify/public_key-jwk`. The JWK path already handles ES256 and even defaults to the IAP key URL for it, so an issuer option (for example `Builder::with_issuers`) is the only thing keeping IAP verification from working today. We verify IAP assertions with jsonwebtoken directly until then.
